### PR TITLE
Upload workflow status as Prow artifacts

### LINKED ISSUE #331: Upload workflow YAML specs as prow artifacts
It would be valuable to upload the Argo YAML spec/status as prow artifacts. This way they would be readily available.

From the YAMl spec users can get the pod id for each step which allows them to fetch the logs from stackdriver.

### PR DESCRIPTION
**Pull request by derekhh**
**Background**

This addresses the issue in #331.

**Changes**

* Dumped workflow results into YAML strings and uploaded them to GCS.

**Tests**

* N/A.

Fix #331 

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/testing/360)

### CODE REVIEW

**Comment by jlewi** [comment]
Thanks
/ok-to-test

**Comment by jlewi** [comment]
Looks great. We just need to wait for the tests to run so that we can verify that the artifacts were uploaded correctly.

File: py/kubeflow/testing/run_e2e_workflow.py [hunk 1]
```diff
@@ -274,6 +275,7 @@ def run(args, file_handler): # pylint: disable=too-many-statements,too-many-bran
       phase = r.get("status", {}).get("phase")
       name = r.get("metadata", {}).get("name")
       workflow_phase[name] = phase
+      workflow_status_yamls[name] = yaml.dump(r)
```
Comment (jlewi) [inline py/kubeflow/testing/run_e2e_workflow.py]: Should we pretty print or otherwise format the YAML to make it human readable?

**Comment by jlewi** [comment]
Could you update the README as well here:
https://github.com/kubeflow/testing/#no-logs-in-argo-ui-for-step-or-pod-id-missing-in-argo-logs

To explain how we can get the pod id from the uploaded Argo spec?

**Comment by jlewi** [comment]
The file was uploaded
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubeflow_testing/360/kubeflow-testing-presubmit/1118924263684837376/kubeflow-testing-presubmit-unittests-360-363f5ad-7376-5ad2.yaml

But it looks like there are some formatting issues. The output looks like

```
    !!python/unicode 'kubeflow-testing-presubmit-unittests-360-363f5ad-7376-5ad2-1967753020': {
```

I think you want to use safe_dump
https://stackoverflow.com/questions/1950306/pyyaml-dumping-without-tags

**Comment by derekhh** [comment]
Got you - I'll fix it. Thanks :)

**Comment by derekhh** [comment]
@jlewi 

Thanks for the feedback. I've updated the PR and verified that the uploaded Prow artifacts look good: https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubeflow_testing/360/kubeflow-testing-presubmit/1120174107468500992/kubeflow-testing-presubmit-unittests-360-105e7c7-0992-1026.yaml

I've also updated the corresponding README section based on your feedback to indicate that we can also retrieve the workflow YAMLs from uploaded Prow artifacts. I also made some Markdown cosmetic updates based on Markdownlint messages.

PTAL?

**Comment by jlewi** [comment]
This is great.

/lgtm
/approve

**Pull request merged**